keps/sig-multicluster/4322-cluster-inventory/README.md

```diff
@@ -0,0 +1,1010 @@
+
+### Spec
+
+#### Display name
+
+It is a human-readable name of the cluster set by the consumer of the
+cluster.
+
+#### Cluster Manager
+
+An immutable field set by a cluster manager when this cluster resources
+is created by it. Each cluster manager instance should set a different
+values to this field.
+
+In addition, a predefined label with key "x-k8s.io/cluster-manager" needs
+to be added by the cluster manager upon creation. The value of the label
+MUST be the same as the name of the cluster manager. The purpose of this
+label is to make filter clusters from different cluster managers easier.
+
+### Status
+
+#### Version
+
+Kubernetes version of the cluster
+
+Versions of the kubernetes can let consumers understand the capability
+of the kubernetes, such as what API is supported.
+
+With recent conversations about kube-apiserver and enabled featureset version,
+it is possible to incorporate other version relating to the cluster, such as
+minimum kubelet version, maximum kubelet version, and enabled featureset version.
+
+#### Resources
+
+Resources include capacity and allocatable, which is the sum of the
+capacity/allocatable of nodes on the cluster. It is useful for users to
+know the sizing of the cluster, and also schedulers to sort or bin pack
+the workload to clusters.
+
+#### Properties
+
+Name/value pairs to represent properties of the clusters. It could be a
+collection of ClusterProperty resources, but could also be info based on
+other implementations.
+
+#### Conditions
+
+Record cluster’s healthiness condition and easy to extend, conforming to
+metav1.Condition format.
+
+Predefined condition types:
+- Healthy conditions indicate the cluster is in a good state. (state:
+  True/False/Unknown). Healthiness can have different meanings in
+  different scenarios. We will have multiple condition types to define
+  healthiness:
+  - ControlPlaneHealthy is to define whether the controlplane of the
+    cluster is in the healthy state.
+    * apiserver/healthz
+    * controller-manager/healthz
+    * scheduler/healthz
+    * etcd/healthz
+  - <<[UNRESOLVED]>> AllNodesHealthy is to define if the nodes in the cluster are in a
+    healthy state. If one node is not healthy, the status of NodeHealthy
+    is set to false with the message indicating details. (todo
+    tolerance, it should be configurable)
+
+  It would be useful to collect the healthiness of other subsystems in
+  the cluster, e.g. network, dns, storage, or ingress. However, it is
+  not easy to collect that information in a common way with different
+  implementations of network or storage providers. We decide not to
+  include other subsystems healthiness conditions in the initial phase.
+- Joined: indicate the cluster is under management by the cluster manager.
+  The status of the cluster SHOULD be updated by the cluster manager under
+  this condition.
+
+## API Example
+
+```yaml
+apiVersion: multicluster.x-k8s.io/v1alpha1
+kind: ClusterInventory
```

Comment:
This Kind looks odd to me. In plain English, "inventory" is a _list of_ items, it is not an item in that list. Wouldn't something like `Cluster` be more appropriate here?